main/web/src/main/java/ru/practicum/explorewithme/main/web/common/mapping/CommentMapper.java

```
@@ -0,0 +1,21 @@
+package ru.practicum.explorewithme.main.web.common.mapping;
+
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Component;
+import ru.practicum.explorewithme.main.model.Comment;
+import ru.practicum.explorewithme.main.web.common.message.CommentDto;
+
+@Component
+@AllArgsConstructor
+public class CommentMapper {
+
+    private final UserMapper userMapper;
+
+    public CommentDto toDto(Comment comment) {
+        return CommentDto.builder()
```

сюда может прилететь null

---

Ну и пусть, в чём проблема? Если передать ноль, то оно и не должно работать. Упадёт сразу, всё по правилу fail fast.

---

так он у тебя 500 ошибку вывалит

---

Ну и пусть. Значит криво написан код. По хорошему тут null не должен быть, так зачем мне отвечать клиенту 4хх, если это явно проблема сервера?

---

ну можно тогда отловить null в лог записать, код тогда конечно криво написан если null туда прилетает

---

Ты предлагаешь в маппере логировать нули? Это ж сугубо внутренний класс, утилитный, который вызывается уже когда даже сервисный слой отработал. Чтобы сервис вернул ноль - это что-то крайне странное. Такие случаи скорее там логироваться и будут.